backend/app/api/api_v1/endpoints/job.py

```diff
@@ -26,6 +29,30 @@ def get_all_job(
         )
     return jobs
 
+@router.get("/skills")
+def get_skills_for_all_jobs(
+    db: Session = Depends(deps.get_db),
+) -> Any:
+    """
+    Get all skills for all jobs.
+    """
+    sql_query = text(
+    '''
+    SELECT 
+        j.Job_ID, j.Job_Name, j.Job_Desc, j.Is_Active as is_job_active, 
+        s.Skill_ID, s.Skill_Name, s.Skill_Desc, s.Is_Active as is_skill_active  
+    FROM job as j
+    LEFT JOIN job_skill ON j.job_ID = job_skill.job_ID
+    LEFT JOIN skill as s ON job_skill.Skill_ID = s.Skill_ID;
+    ''')
+    jobs_with_skills = db.execute(sql_query)
+    if not jobs_with_skills:
+        raise HTTPException(
+            status_code=404,
+            detail="Error getting all jobs with their skills",
```

Comment:
/nit rather than error, would it be better to respond that there is no information available?

Reply:
I think if there is no info there will still be the cursor object then I'll return [] since jobs_with_skills is a Cursor. So if no cursor is returned means that there is an error.

Reply:
O wait its changed now. Hmm if we do db.execute().all(), if the db.execute() fails, will the .all chain fail?

Reply:
Was referring to the error message actually